### ISSUE TITLE
Duplicated exercise in topic tree

### ISSUE THREAD

**Issue by rtibbles** [learningequality/ka-lite#4396]
For some reason, we have two almost completely identical exercises in the topic tree (I believe this is probably due to an editing error on KA's part, where the data has been inserted twice under two slightly different names).

This causes us to have non-unique paths.

Data below:
```
[{'all_assessment_items': ['{"sha": "d124afecdeb491da7ef297aafde30f824d142309", "live": true, "id": "x27af6ca3e3dd6586", "perseus_api_major_version": 3, "requires_screen_or_mouse": false}',
   '{"sha": "e71f206d4e3b0b4fc47d62a5b06244d4df34918b", "live": true, "id": "x503f312d62af87a3", "perseus_api_major_version": 3, "requires_screen_or_mouse": false}',
   '{"sha": "488f2b1dedb412b6a603b2434fcc4af9b4380481", "live": true, "id": "xa11ed08fb084c2fd", "perseus_api_major_version": 3, "requires_screen_or_mouse": false}',
   '{"sha": "e246cb8cc6e4b79a645278d0d9d0646b295b73a7", "live": true, "id": "x861e7f6afd36266d", "perseus_api_major_version": 3, "requires_screen_or_mouse": false}',
   '{"sha": "3bc3136183c1490e9b76a96121d2112dad36484a", "live": true, "id": "xf0f42f8dcea44fed", "perseus_api_major_version": 3, "requires_screen_or_mouse": false}'],
  'available': False,
  'basepoints': 10.0,
  'description': 'Passages that test your knowledge on fluids in motion',
  'display_name': 'A fluid flowing through a tube',
  'exercise_id': 'fluids-in-motion---passage-1',
  'id': 'fluids-in-motion---passage-1',
  'kind': 'Exercise',
  'name': 'fluids-in-motion---passage-1',
  'path': 'khan/test-prep/mcat/physical-sciences-practice/physical-sciences-practice-tut/fluids-in-motion-passage-1/',
  'prerequisites': [],
  'seconds_per_fast_problem': 4.0,
  'slug': 'fluids-in-motion-passage-1',
  'title': 'A fluid flowing through a tube',
  'uses_assessment_items': True},
 {'all_assessment_items': ['{"sha": "11e79a919a2dd45b896dcf94ac816bfc8ad1d242", "live": true, "id": "xf62865128a3d3ae7"}',
   '{"sha": "3e72694621a9ef28df0f3ea8cfaa03d761fd8733", "live": true, "id": "x122f1a1565828399"}',
   '{"sha": "693f2c8e84ea4f571915d86d620872c40f8dcad0", "live": true, "id": "xfcf8ba8dee680e66"}',
   '{"sha": "cbca1e2fb3b5f5ca254b2e49d4a88aa098cc763c", "live": true, "id": "x2826ef9efce4d735"}',
   '{"sha": "4c51b93a5b11f1bf97d16ec419a971bd3dea4ea6", "live": true, "id": "x3b3892b3d539ad2c"}'],
  'available': False,
  'basepoints': 10.0,
  'description': 'This passage will test your knowledge on fluids in motion',
  'display_name': 'The tubular system in an aquarium',
  'exercise_id': 'fluids-in-motion--passage-1',
  'id': 'fluids-in-motion--passage-1',
  'kind': 'Exercise',
  'name': 'fluids-in-motion--passage-1',
  'path': 'khan/test-prep/mcat/physical-sciences-practice/physical-sciences-practice-tut/fluids-in-motion-passage-1/',
  'prerequisites': [],
  'seconds_per_fast_problem': 4.0,
  'slug': 'fluids-in-motion-passage-1',
  'title': 'The tubular system in an aquarium',
  'uses_assessment_items': True}]
```

**Comment by rtibbles** [comment]
To add some more confusion, neither of these exercises actually appear on Khan Academy.

**Comment by rtibbles** [comment]
Seems like this are, in fact, distinct exercises. Issue arises from the fact that the slugs are rendered identical by Django slugify stripping out repeated ```-```.

**Comment by rtibbles** [comment]
Fixed in #4397

**Comment by rtibbles** [comment]
@MCGallaspy reports new issues.

**Comment by MCGallaspy** [comment]
This path is duplicated: /learn/khan/math/early-math/cc-early-math-measure-data-topic/cc-early-math-length-intro/order-by-length/

It's both an exercise and a video.

This one too: /learn/khan/math/early-math/cc-early-math-geometry-topic/cc-early-math-fractions-of-shapes/equal-parts-of-circles-and-rectangles/

**Comment by MCGallaspy** [comment]
To be clear, clicking the sidebar entry for either of those two items will only bring you to one -- the other is not accessible.

**Comment by rtibbles** [comment]
Figured this out - in 'fixing' the deduplication code, I inadvertently prevented it from recursing the whole topic tree, so it was only deduplicating the top level. Will put in a PR shortly once I have updated the topic tree and contents.

**Comment by rtibbles** [comment]
Assuming fixed by #4422 - if you see a recurrence, @MCGallaspy, please feel free to reopen.
